tools/psa/__init__.py

```diff
@@ -0,0 +1,17 @@
+from os.path import basename
+
+
+# Find secure image.
+def find_secure_image(toolchain, resources, ns_image_path, configured_s_image_path, image_type):
+    # Locate user-specified image
+    image_files = resources.get_file_paths(image_type)
+    secure_image = next((f for f in image_files if basename(f) == configured_s_image_path), None)
+    secure_image = next((f for f in image_files if basename(f) == basename(ns_image_path)), secure_image)
+
+    if secure_image:
+        toolchain.notify.debug("Secure image file found: %s." % secure_image)
```

Comment:
Resources has a notifier too, and if you're only using the notifier from the toolchain, pass in the notifier please.

Reply:
Ok

Reply:
Well it didn't work for resources notifier:
```sh
Traceback (most recent call last):
  File "C:\miki_dev\mbed-os_backup\tools\test_api.py", line 2210, in build_test_worker
    bin_file, _ = build_project(*args, **kwargs)
  File "C:\miki_dev\mbed-os_backup\tools\build_api.py", line 584, in build_project
    res, _ = toolchain.link_program(resources, build_path, name)
  File "C:\miki_dev\mbed-os_backup\tools\toolchains\__init__.py", line 654, in link_program
    self.binary(r, elf, bin)
  File "C:\miki_dev\mbed-os_backup\tools\hooks.py", line 54, in wrapper
    post_res = tooldesc["post"](t_self, *args, **kwargs)
  File "C:\miki_dev\mbed-os_backup\tools\targets\__init__.py", line 604, in binary_hook
    secure_bin = find_secure_image(t_self, resources, binf, configured_secure_image_filename, FileType.BIN)
  File "C:\miki_dev\mbed-os_backup\tools\psa\__init__.py", line 12, in find_secure_image
    resources.notify.debug("Secure image file found: %s." % secure_image)
AttributeError: 'Resources' object has no attribute 'notify'

```

Reply:
Probably private then. Pass in a notifier instead.